ingest/expression_files/dense_ingestor.py

```diff
@@ -228,19 +230,26 @@ def transform(self):
                     gene, id, numeric_scores
                 ):
                     data_arrays.append(gene_expression_values)
-                if len(gene_models) == 5:
-                    num_processed += len(gene_models)
-                    self.info_logger.info(
-                        f"Processed {num_processed} models, "
-                        f"{str(datetime.datetime.now() - start_time)} elapsed",
-                        extra=self.extra_log_params,
-                    )
-                    yield gene_models, data_arrays
-                    gene_models = []
-                    data_arrays = []
-        yield gene_models, data_arrays
-        num_processed += len(gene_models)
-        self.info_logger.info(
-            f"Processed {num_processed} models, {str(datetime.datetime.now() - start_time)} elapsed",
-            extra=self.extra_log_params,
-        )
+            if (
+                len(gene_models) >= self.GENE_BATCH_SIZE or
+                len(data_arrays) >= self.DATA_ARRAY_BATCH_SIZE
+            ):
+                num_processed += len(gene_models)
+                self.info_logger.info(
+                    f"Processed {num_processed} models, "
+                    f"{str(datetime.datetime.now() - start_time)} elapsed",
+                    extra=self.extra_log_params,
+                )
+                yield gene_models, data_arrays
+                gene_models = []
+                data_arrays = []
+
+        # load any remaining models (this is necessary here since there isn't an easy way to detect the
+        # last line of the file in the iteration above
+        if len(gene_models) > 0:
```

Comment:
Non blocker but this can be ``` if gene_model```

Reply:
I can see this going either way -- `if gene_model` might be more pythonic, but in critical parsing code I'd rather be really explicit about what we are testing.  it's too easy to forget what counts as truthy in Python/Ruby/Javascript (e.g. in javascript `if ([])` evaluates as true) and have bugs creep in as a result